lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb

```diff
@@ -136,6 +138,35 @@ def self.print_override_warning(pod_bundle, target, config, key)
               'This can lead to problems with the CocoaPods installation'
             UI.warn(message, actions)
           end
+
+          # Naively checks to see if a given PBXFileReference imports a given
+          # path.
+          #
+          # @param  [PBXFileReference] base_config_ref
+          #         A file reference to an `.xcconfig` file.
+          #
+          # @param  [String] target_config_path
+          #         The path to check for.
+          #
+          SILENCE_WARNINGS_STRING = '// @COCOAPODS_SILENCE_WARNINGS@ //'
+          def self.xcconfig_includes_target_xcconfig?(base_config_ref, target_config_path)
+            return false unless base_config_ref && File.exist?(base_config_ref.real_path)
+            regex = /
+              ^(
+                (\s*                                  # Possible, but unlikely, space before include statement
+                  \#include\s+                        # Include statement
+                  ['"]                                # Open quote
+                  (.*\/)?                             # Possible prefix to path
+                  #{Regexp.quote(target_config_path)} # The path should end in the target_config_path
+                  ['"]\s*                             # Close quote and potential excess whitespace
+                )
+                |
+                (#{Regexp.quote(SILENCE_WARNINGS_STRING)}) # Token to treat xcconfig as good and silence pod install warnings
+              )$
+            /x
+            File.foreach(base_config_ref.real_path) { |line| return true if regex.match line }
```

Comment:
Ah, yeah `=~` is better. I went for `foreach` because there's no need to read the entire file into memory here—the `#import` statements will all be at the beginning.